### PR TITLE
Document how to setup Makefile for 3rd party boards

### PR DESCRIPTION
**Pull request by MilanV**
Add a documentation for the others not to fall into the #499 issue

### CODE REVIEW

**Comment by sudar** [comment]
@sej7278 

What are your thoughts on this PR?

File: examples/MakefileExample/Makefile-3rd_party-board.mk [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+###        │       ├── bootloaders/
+###        │       ├── driver/
+###        │       ├── platform.txt
+###        │       ├── signed_driver/
+###        │       └── variants/
+
+### 3. Create this Makefile (use your vendor/package name)
+
+###        ALTERNATE_CORE      = sparkfun
+###        include ~/Arduino-Makefile/Arduino.mk
```
Comment (sej7278) [inline examples/MakefileExample/Makefile-3rd_party-board.mk]: don't use tilde

File: examples/MakefileExample/Makefile-3rd_party-board.mk [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#####
+
+### The basic configuration should be done now.
+### The example follows:
+
+ARDUINO_SKETCHBOOK  = $(HOME)/sketchbook
+ALTERNATE_CORE      = sparkfun
+BOARD_TAG           = promicro
+BOARD_SUB	    = 8MHzatmega32U4
+include $(ARDMK_DIR)/Arduino.mk
```
Comment (sej7278) [inline examples/MakefileExample/Makefile-3rd_party-board.mk]: ARDMK_DIR isn't defined, can we use an absolute path e.g /usr/share/arduino/Arduino.mk ?

**Comment by MilanV** [comment]
Thanks for comments, I fixed it.

**Comment by sej7278** [comment]
Looks good to me now thanks! :+1: 

People aren't looking at https://github.com/sudar/Arduino-Makefile/blob/master/examples/Blink/Makefile i guess, as that has various board examples too.....

**Comment by MilanV** [comment]
I still see a room for improvement here. I believe a new document as a root of the documentation would be helpful. The one describing the whole ecosystem: terminology, list of config files and which does what, what is used from the Arduino IDE etc. Then, the document I created would become just one chapter ("Using a board not defined in he standard Arduino package"). For example, I found a description of Makefile targets only thanks to a TAB completion on Linux command-line, which suggested "help" as one of possible targets of this Makefile :)

But that is definitely out of my knowledge, I'm still an arduino greenhorn :) Just spreading the idea...

**Comment by sej7278** [comment]
Yes, we could probably make a whole .MD file about 3rd party boards, merging now thanks

**Comment by zvipesh** [comment]
WINDOWS-7:

Is it possible to list the required PATH setting for the Arduino IDE 1.8.2 and any other tool required in order to generate the "bootloaderxxx.hex" ?

Do we need to install additional tools besides the IDE installed files ?

I keep getting error indicating the MAKE function does not finds the destination file .
something like the following will be of great help.

for example:
1. Install IDE 1.8.2 in c:\program files (x86)
2. Set PATH as follows:
   a. in control panel - Control Panel\System and Security\System  click the "Advanced system setting"
   b. Click the tab : Environment variables"  
   c. In the "user variables for xx" select  PATH and click EDIT tab
   d.  Add   to PATH:  C:\Program Files (x86)\Arduino\hardware\tools\avr\bin;
         : 
         : < list all PATHs needed to run the compiler etc >
   f.  In the "system variables" select  PATH and click EDIT tab
   g.  Add   to PATH:   < List PATH parameters >

   h. when done - press OK
   i.  Press OK again in the System Properties.

my "user variables"  PATH parameters are:
C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\GnuWin32\bin;C:\Program Files (x86)\AVRDUDE 6.3;C:\Program Files (x86)\Arduino;c:\Program Files (x86)\Arduino\hardware\arduino\avr\bootloaders\stk500v2;

and the "system variables are:
C:\Program Files (x86);C:\Program Files (x86)\GnuWin32\bin;C:\Program Files (x86)\Arduino\hardware\tools\avr\bin;C:\WinAVR-20100110\utils\bin;C:\Program Files (x86)\AVRDUDE 6.3;

It all worked  until I had a window update yesterday , tried several iterations to fix the PATH  ..
no luck ..

this is why I suggest there should be a "specific guide" since the MAKE function is not working
when trying for example ( in installed  IDE 1.8.2  ) when in the c:\...\arduino\hardware\arduino\avr\bootloaders\stk500v2 folder and executing MAKE
it does not works. < tried it after a new fresh fresh installation of the IDE >

make_bootloader generation 6-20-2017

**Comment by sej7278** [comment]
We don't really support windows (it does work apparently but we don't really have a way to test it) but would appreciate a PR if someone fancies documenting how to do it.

**Comment by zvipesh** [comment]
what do you support ?  is it lynix ? OS-x ?

**Comment by sej7278** [comment]
linux and osx are the platforms that most of the contributors to this project use, windows apparently works, as do the bsd's. i suggest you search the issues as there are other windows users who have commented.

**Pull request merged**